File: 0x08-python-more_classes/3-rectangle.py
#!/usr/bin/python3
""" Creating a class called Rectangle"""


class Rectangle:
    """Make rectangle object"""

    def __init__(self, width=0, height=0):
        """Make a new rectangle object"""
        self.height = height
        self.width = width

    @property
    def width(self):
        """gets width of the rectangle"""
        return self.__width

    @width.setter
    def width(self, value):
        """sets width of the rectangle"""
        if not isinstance(value, int):
            raise TypeError("width must be an integer")
        elif value < 0:
            raise ValueError("width must be >= 0")
        self.__width = value

    @property
    def height(self):
        """gets height of the rectangle"""
        return (self.__height)

    @height.setter
    def height(self, value):
        """sets height of the rectangle"""
        if not isinstance(value, int):
            raise TypeError("height must be an integer")
        elif value < 0:
            raise ValueError("height must be >= 0")
        self.__height = value

    def area(self):
        """Return areas of a rectangle."""
        return (self.__width * self.__height)

    def perimeter(self):
        """Returns perimeter of a rectangle"""
        if self.__height == 0 or self.__width == 0:
            return 0
        return ((self.__width + self.__height) * 2)

    def __str__(self):
        """Return the printable representation of the Rectangle.
        Represents the rectangle with the # character.
        """
        if self.__width == 0 or self.__height == 0:
            return ("")

        rect = []
        for i in range(self.__height):
            [rect.append('#') for j in range(self.__width)]
            if i != self.__height - 1:
                rect.append("\n")
        return ("".join(rect))

File: 0x13-javascript_objects_scopes_closures/tests/3-rectangle.js
#!/usr/bin/node
module.exports = class Rectangle {
  constructor (w, h) {
    if (!Number.isInteger(w) || w <= 0 || !Number.isInteger(h) || h <= 0) {
      let Rectangle = {};
      return Rectangle.name;
    }
    this.width = w;
    this.height = h;
  }
  print () {
   for (let i = 0; i < this.height; i++) {
    let row = "";
    for (let j = 0; j < this.width; j++) {
      row += "X";
    }
    console.log(row);
   } 
 } 
};
